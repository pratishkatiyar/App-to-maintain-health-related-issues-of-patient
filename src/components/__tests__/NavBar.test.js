import { render, screen, cleanup } from '@testing-library/react'
import { BrowserRouter, Route, Switch } from 'react-router-dom';
import Navbar from '../Navbar'
import renderer, { ReactTestRendererJSON } from 'react-test-renderer';

afterEach(() => {
    cleanup();
})

test('should render Navbar component', () =>{
    render(<BrowserRouter><Navbar/></BrowserRouter>);
    const navBarElement = screen.getByTestId('navbar-1');
    expect(navBarElement).toBeInTheDocument();
    expect(navBarElement).toHaveTextContent('Patient Registration App');
})

test('matches snapshot', ()=> {
    const tree = renderer.create(<BrowserRouter><Navbar/></BrowserRouter>).toJSON();
    expect(tree).toMatchSnapshot();
})