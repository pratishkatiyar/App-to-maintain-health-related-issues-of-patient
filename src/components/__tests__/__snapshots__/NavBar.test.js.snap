// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  data-testid="navbar-1"
>
  <nav>
    <div
      className="logo"
    >
      <p>
        <a
          href="/"
          onClick={[Function]}
        >
          Home
        </a>
      </p>
    </div>
    <div
      className="logo"
    >
      Patient Registration App
    </div>
    <div />
  </nav>
</div>
`;
